### PR TITLE
PR Preview: Menu Item Availability

### PR DESCRIPTION
**Pull request by tidy-dev**
## Description
This PR makes updates the menu items availability and renames it to "Preview" as opposed to "Start" to have consistent referencing. 

Disabled when:
- another popup is open
- repository not active
- branch tip state is unborn or detached

Available changes:
- unpublished repo (as opposed to creating a pr)
- there is a PR open for the branch (as opposed to creating a pr)

## Release notes
Notes: [Improved] Preview Pull Request menu item is availability is consistent with other menu items.

### CODE REVIEW

File: app/src/main-process/menu/build-default-menu.ts
Comment (tidy-dev) [inline app/src/main-process/menu/build-default-menu.ts:437]: I originally had this `!hasCurrentPullReques` to follow the convention for the create pull request item which switches to an open pull request item in this scenario. But, I don't think there is a good reason to preview a user from seeing the pull request preview of an open PR.

File: app/src/main-process/menu/build-default-menu.ts
Comment (tidy-dev) [inline app/src/main-process/menu/build-default-menu.ts:439]: After doing the suggested next actions changes, I realized that "Start Pull Request" doesn't exactly make sense to differentiate from creating a pull request. Decided to make consistent verbiage of "Preview Pull Request". But, I am completely open to discussion about what terminology would communicate the intention the best.

File: app/src/main-process/menu/build-default-menu.ts
Comment (sergiou87) [reply app/src/main-process/menu/build-default-menu.ts:437]: Wouldn't it be confusing? If there is already a PR, seeing a "Preview PR" could lead you to think there isn't one already 🤔

**Review by sergiou87 — APPROVED** [review]
LGTM! Only have one question, but not gonna block the PR on that 😄

File: app/src/main-process/menu/build-default-menu.ts
Comment (tidy-dev) [reply app/src/main-process/menu/build-default-menu.ts:437]: Create pr would be disabled. This would just be for viewing the PR Diff which I found myself wanting to do after making some changes to existing PR. .. If you think it would be confusing, we could switch the verbage to "View Pull Request Diff" in this scenario. Another thought not necessarily for this PR, but the create button could change to a push button.

**Pull request merged**
